File: app/src/main/java/cn/intret/app/picgo/model/user/SortWay.java
package cn.intret.app.picgo.model.user;

public enum SortWay {
    NAME,
    SIZE,
    DATE,
    UNKNOWN;
    public static SortWay fromString(String string) {
        if (string == null) {
            return UNKNOWN;
        }
        switch (string) {
            case "NAME": return NAME;
            case "SIZE": return SIZE;
            case "DATE": return DATE;
        }
        return UNKNOWN;
    }

    @Override
    public String toString() {
        switch (this) {
            case NAME: return "NAME";
            case SIZE: return "SIZE";
            case DATE: return "DATE";
        }
        return "UNKNOWN";
    }
}

File: app/src/main/java/cn/intret/app/picgo/model/user/data/SortWay.kt
package cn.intret.app.picgo.model.user.data

/**
 * Image list sort way
 */
enum class SortWay {
    NAME,
    SIZE,
    DATE,
    UNKNOWN;

    override fun toString(): String {
        when (this) {
            NAME -> return "NAME"
            SIZE -> return "SIZE"
            DATE -> return "DATE"
        }
        return "UNKNOWN"
    }

    companion object {
        fun fromString(string: String?): SortWay {
            if (string == null) {
                return UNKNOWN
            }
            when (string) {
                "NAME" -> return NAME
                "SIZE" -> return SIZE
                "DATE" -> return DATE
            }
            return UNKNOWN
        }
    }
}
